iceoryx_examples/automotive_soa/include/owl/kom/field_publisher.inl

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_FIELD_PUBLISHER_INL
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_FIELD_PUBLISHER_INL
+
+#include "owl/kom/field_publisher.hpp"
+
+namespace owl
+{
+namespace kom
+{
+template <typename T>
+inline FieldPublisher<T>::FieldPublisher(const core::String& service,
+                                         const core::String& instance,
+                                         const core::String& event,
+                                         FieldType& field) noexcept
+    : m_publisher({service, instance, event}, {HISTORY_CAPACITY})
+    , m_server({service, instance, event})
+    , m_latestValue(field)
+{
+    // publisher is automatically offered
+    Update(field);
+
+    //! [FieldPublisher attach]
+    m_listener
+        .attachEvent(m_server,
+                     iox::popo::ServerEvent::REQUEST_RECEIVED,
+                     iox::popo::createNotificationCallback(onRequestReceived, *this))
+        .or_else([](auto) {
```

Comment:
You can write `expect("unable to attach server");` instead of this `or_else` thingy.